core/app/models/spree/stock_location.rb

```diff
@@ -51,11 +51,14 @@ def set_up_stock_item(variant)
 
     # Returns an instance of StockItem for the variant id.
     #
-    # @param variant_id [String] The id of a variant.
+    # @param variant_or_id [String] The id of a variant.
     #
     # @return [StockItem] Corresponding StockItem for the StockLocation's variant.
-    def stock_item(variant_id)
-      stock_items.where(variant_id: variant_id).order(:id).first
```

Comment:
I can't think of why it matters, but could losing the order on ID matter?

Reply:
Hm, there can only be one undeleted stock item per stock location and variant, so I'd say no:
https://github.com/solidusio/solidus/blob/master/core/app/models/spree/stock_item.rb#L12